hadoop-hdds/rocks-native/src/test/java/org/apache/hadoop/hdds/utils/db/managed/TestManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,243 @@
+            Named.of("Value starting & ending with a number " +
+                "& containing null character & new line character",
+                "%1$dvalue\n\0%1$d")
+        ),
+        Arguments.of(
+            Named.of("Key ending with a number & containing" +
+                " a null character", "key\0%1$d"),
+            Named.of("Value starting & ending with a number " +
+                "& elosed within quotes", "%1$dvalue\r%1$d")
+        )
+    );
+  }
+
+  private static byte[] getBytes(Integer val) {
+    byte[] b = new byte[4];
+    for (int i = 3; i >= 0; i--) {
+      b[i] = val.byteValue();
+      val = val >> 8;
+    }
+    return b;
+  }
+
+  private static byte[] getBytes(Long val) {
+    byte[] b = new byte[8];
+    for (int i = 7; i >= 0; i--) {
+      b[i] = (byte)(val & 0xff);
+      val = val >> 8;
+    }
+    return b;
+  }
+
+  private static byte[] getBytes(String val) {
+    byte[] b = new byte[val.length()];
+    for (int i = 0; i < val.length(); i++) {
+      b[i] = (byte) val.charAt(i);
+    }
+    return b;
+  }
+
+  private static Stream<? extends Arguments> invalidPipeInputStreamBytes() {
+    return Stream.of(
+        Arguments.of(Named.of("Invalid 3 byte integer",
+            new byte[] {0, 0, 0})),
+        Arguments.of(Named.of("Invalid 2 byte integer",
+            new byte[] {0, 0})),
+        Arguments.of(Named.of("Invalid 1 byte integer",
+            new byte[] {0, 0})),
+        Arguments.of(Named.of("Invalid key name length",
+            Bytes.concat(getBytes(4), getBytes("key")))),
+        Arguments.of(Named.of("Invalid Unsigned Long length",
+            Bytes.concat(getBytes(4), getBytes("key1"),
+                new byte[]{0, 0}))),
+        Arguments.of(Named.of("Invalid Sequence number",
+            Bytes.concat(getBytes(4), getBytes("key1")))),
+        Arguments.of(Named.of("Invalid Type",
+            Bytes.concat(getBytes(4), getBytes("key1"),
+                getBytes(4L)))),
+        Arguments.of(Named.of("Invalid Value",
+                Bytes.concat(getBytes(4), getBytes("key"),
+                    getBytes(4L), getBytes(0)))),
+        Arguments.of(Named.of("Invalid Value length",
+            Bytes.concat(getBytes(4), getBytes("key"),
+                getBytes(4L), getBytes(1), getBytes(6),
+                getBytes("val"))))
+    );
+  }
+
+
+
+  @Native("Managed Rocks Tools")
+  @ParameterizedTest
+  @MethodSource("keyValueFormatArgs")
+  public void testSSTDumpIteratorWithKeyFormat(String keyFormat,
+              String valueFormat) throws Exception {
+    TreeMap<Pair<String, Integer>, String> keys =
+        IntStream.range(0, 100).boxed().collect(
+            Collectors.toMap(
+                i -> Pair.of(String.format(keyFormat, i), i % 2),
+                i -> i % 2 == 0 ? "" : String.format(valueFormat, i),
+                (v1, v2) -> v2,
+                TreeMap::new));
+    testSSTDumpIteratorWithKeys(keys);
```

Comment:
It is not a good idea to delegate one test:`testSSTDumpIteratorWithKeyFormat` on another: `testSSTDumpIteratorWithKeys`.

If you want, can divide the big test into parts like:

```
public void test() {
  - Given
  // Test setup and mocking. Can be another method, if needed.
  - When
  //  method call writing test for. Can be another method, if needed.
  - Then
  // Assertions and verifications. Can be another method, if needed.
}
```

Reply:
done